### PR TITLE
rework package

### LINKED ISSUE #1: Make it flutter web ready

### PR DESCRIPTION
**Pull request by ahmednfwela**
fixes #1 

- Reworked package based on the latest flutter/dart versions
- Added `google_fonts` as a dependency to solve web compatibility 
- Added `iconOnly` variant, that displays the BMC icon instead of the entire row

### CODE REVIEW

**Comment by ahmednfwela** [comment]
cc @md-weber

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (md-weber) [inline .vscode/launch.json]: NIT: Could we add .vscode into the `.gitignore` file.

**Comment by md-weber** [comment]
Thanks so much for your contribution! Most of it looks ok, just some questions:
- Why did you remove the android example?

Addition, we would probably to add to the readme that we use the GoogleFonts package because it would clash with the GDPR I believe (AFAIK it fetches the Fonts via Web call).

**Comment by ahmednfwela** [comment]
oh I wanted to generate the example files but forgot ,sorry about that.

also Google fonts doesn't need to fetch fonts since we add them to assets (according to their docs)

File: .vscode/launch.json (hunk 1 above)
Comment (ahmednfwela) [reply .vscode/launch.json]: done

**Comment by md-weber** [comment]
Tested on:
- [x] iOS
- [x] Android
- [x] Web

LGTM, great job, thanks again for your contribution!

**Pull request merged**